project/cmake/scripts/common/addon-helpers.cmake

```diff
@@ -137,6 +140,20 @@ macro (build_addon target prefix libs)
     INSTALL(TARGETS ${target} DESTINATION ${CMAKE_INSTALL_LIBDIR}/addons/${target})
     INSTALL(DIRECTORY ${target} DESTINATION share/${APP_NAME_LC}/addons PATTERN "addon.xml.in" EXCLUDE)
   ENDIF(PACKAGE_ZIP OR PACKAGE_TGZ)
+  if(KODI_BUILD_DIR)
+    file(GLOB_RECURSE files ${CMAKE_CURRENT_SOURCE_DIR}/${target}/*)
+    foreach(file ${files})
+      string(REPLACE "${CMAKE_CURRENT_SOURCE_DIR}/${target}/" "" name "${file}")
+      # A good way to deal with () in filenames
+      if(NOT ${file} MATCHES addon.xml.in)
+        configure_file(${file} ${KODI_BUILD_DIR}/addons/${target}/${name} COPYONLY)
+      endif()
+    endforeach()
+    add_custom_command(TARGET ${target} POST_BUILD
+                       COMMAND ${CMAKE_COMMAND} -E copy
+                                $<TARGET_LINKER_FILE:${target}>
+                                ${KODI_BUILD_DIR}/addons/${target}/$<TARGET_LINKER_FILE_NAME:${target}>)
```

Comment:
Likely a good idea.
23. apr. 2016 16:38 skrev "Christian Fetzer" <notifications@github.com>:

> In project/cmake/scripts/common/addon-helpers.cmake
> <https://github.com/fetzerch/xbmc/pull/2#discussion_r60829441>:
>
> > @@ -137,6 +140,20 @@ macro (build_addon target prefix libs)
> >      INSTALL(TARGETS ${target} DESTINATION ${CMAKE_INSTALL_LIBDIR}/addons/${target})
> >      INSTALL(DIRECTORY ${target} DESTINATION share/${APP_NAME_LC}/addons PATTERN "addon.xml.in" EXCLUDE)
> >    ENDIF(PACKAGE_ZIP OR PACKAGE_TGZ)
> > +  if(KODI_BUILD_DIR)
> > +    file(GLOB_RECURSE files ${CMAKE_CURRENT_SOURCE_DIR}/${target}/*)
> > +    foreach(file ${files})
> > +      string(REPLACE "${CMAKE_CURRENT_SOURCE_DIR}/${target}/" "" name "${file}")
> > +      # A good way to deal with () in filenames
> > +      if(NOT ${file} MATCHES addon.xml.in)
> > +        configure_file(${file} ${KODI_BUILD_DIR}/addons/${target}/${name} COPYONLY)
> > +      endif()
> > +    endforeach()
> > +    add_custom_command(TARGET ${target} POST_BUILD
> > +                       COMMAND ${CMAKE_COMMAND} -E copy
> > +                                $<TARGET_LINKER_FILE:${target}>
> > +                                ${KODI_BUILD_DIR}/addons/${target}/$<TARGET_LINKER_FILE_NAME:${target}>)
>
> @notspiff <https://github.com/notspiff>: I'm wondering if it wouldn't be
> better to copy the TARGET_FILE (so.X.X.X) to match the name in the
> addon.xmls. (f.e. library_linux="audiodecoder.timidity.so.1.0.3").
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/fetzerch/xbmc/pull/2/files/4828998ebbd532901f7047cc46f0842f92d09879#r60829441>
>